x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/azureopenai/embeddings/AzureOpenAiEmbeddingsServiceSettings.java

```diff
@@ -261,6 +293,7 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeBoolean(dimensionsSetByUser);
         out.writeOptionalVInt(maxInputTokens);
         out.writeOptionalEnum(SimilarityMeasure.translateSimilarity(similarity, out.getTransportVersion()));
+        rateLimitSettings.writeTo(out);
```

Comment:
Similar approach as the constructor here, I'm hiding the `if-block` check for the transport version within the settings `writeTo` method. Let me know you think that's fine.

Reply:
LGTM